internal/background/availability_queue.go

```diff
@@ -31,16 +42,40 @@ func sendAvailabilityRequestMessages(ctx context.Context, batchSize int, tickDur
 	logger := ctxval.Logger(ctx)
 	ticker := time.NewTicker(tickDuration)
 	messageBuffer := make([]*kafka.GenericMessage, 0, batchSize)
+	sc, err := clients.GetSourcesClient(ctx)
+	if err != nil {
+		logger.Warn().Msg("Could not get sources client")
+		return
+	}
 
 	for {
 		select {
-		case msg := <-kafkaAvailabilityRequest:
-			messageBuffer = append(messageBuffer, msg)
+		case info := <-kafkaAvailabilityRequest:
+			newCtx, err := ctxval.WithIdentityFrom64(ctx, info.identity)
+			if err != nil {
+				logger.Warn().Msg("Could not add identity header to ctx")
+				return
```

Comment:
Should this be "continue"? Would it be helpful to include the error here as well? It will be easier to debug later on ;)
```suggestion
				logger.Warn().Err(err).Msg("Could not add identity header to ctx")
				continue
```